geoportal/c2cgeoportal_geoportal/lib/oidc.py

```diff
@@ -194,6 +210,14 @@ def get_user_from_remember(
                 user = static.User(username=username, email=email, display_name=display_name)
                 models.DBSession.add(user)
     else:
+        roles = []
+        role_names = remember_object.get("roles", [])
+        if role_names:
+            query = models.DBSession.query(main.Role).filter(main.Role.name.in_(role_names))
+            if openid_connect_configuration.get("ignore_unknown_roles", False):
+                roles = [role for role in query.all() if role is not None]
```

Comment:
`models.DBSession.query(main.Role).filter(main.Role.name.in_(role_names))` could return some `None`?

Not sure to understand option `ignore_unknown_roles` if you request roles from `main.role` with an `IN` filter.

Reply:
Good question, I have to remember that I get some None but your right it has no sense!
I will remove it and see what's happens :-)